### ISSUE TITLE
Weird customs font rendering

### ISSUE THREAD

**Issue by SKProCH** [GTNewHorizons/Angelica#285]
### Your GTNH Discord Username

skproch

### Mod Version

1.0.0-alpha25

### Java Version

Java 21

### Bug Report

When I using texturepack with custom fonts seems like fonts rendering blurry and strange under/upperscores appeared
Also, optifine compatible fonts seems like doesn't work

Here is the images:  
![image](https://github.com/GTNewHorizons/Angelica/assets/29896317/a146a6bc-00f3-49cc-b719-c1523b2a0c48)
![image](https://github.com/GTNewHorizons/Angelica/assets/29896317/b3f015a8-d08d-4240-8ae9-17c1f18005ca)
(zoom images to see upper/undedscores)

It's not related to multiplayer menu, it renders weirdly everywhere.

Without resourcepacks it renders properly:  
![image](https://github.com/GTNewHorizons/Angelica/assets/29896317/dc08f8cd-8c7c-47f3-9446-b1c17939f3db)

Resourcepacks is [Dunamic Duo General.zip](https://github.com/GTNewHorizons/Angelica/files/14234338/Dunamic.Duo.General.zip) and [Dunamic Duo RUS.zip](https://github.com/GTNewHorizons/Angelica/files/14234340/Dunamic.Duo.RUS.zip). Rus must be above general in the list.

Same resourcepacks but with optifine look fine: 
![image](https://github.com/GTNewHorizons/Angelica/assets/29896317/27e08bf2-942f-43aa-9ce5-b8aeddfe289a)
Without Optifine and angelica it renders as without resourcepacks.

TL;DR: Seems like angelica doesn't support optifine compatible resourcepacks and using then gives a strange under/upperscores while vanilla does render them properly.

### Mod List or GTNH Pack Version

GTNH 2.5.1

### Final Checklist

- [X] I have searched the issues and haven't found a similar issue.
- [X] I have read the known incompatibilities and this is not related to one of those.
- [X] I am running an officially released version.  (Or, if I've compiled it myself I plan to fix the issue)
- [X] This issue is not related to a feature that is disabed by default - Shaders, MCPF, etc.  [They'll be enabled when they're ready for testing]

**Comment by mitchej123** [comment]
Does this happen every time?  Do you have a link to such a resource pack?

The biggest trouble I've seen is thus far it's been intermittent at best for me and impossible to reliably reproduce...

**Comment by SKProCH** [comment]
Yep, it's constant problem - happens every time (seems like).

Also, I've linked resourcepacks in my first message 
> Resourcepacks is [Dunamic Duo General.zip](https://github.com/GTNewHorizons/Angelica/files/14234338/Dunamic.Duo.General.zip) and [Dunamic Duo RUS.zip](https://github.com/GTNewHorizons/Angelica/files/14234340/Dunamic.Duo.RUS.zip). Rus must be above general in the list.

**Comment by mitchej123** [comment]
Does it still happen if you run with `-Dangelica.disableGlCache=true` as a JVM arg?

> Also, I've linked resourcepacks in my first message

Missed it in the wall of images :sunglasses:

**Comment by SKProCH** [comment]
Adding `-Dangelica.disableGlCache=true` resolves the problem, and fonts starts rendering without under/upperscores and more clearly.

**Comment by mitchej123** [comment]
What graphics card are you running?

**Comment by SKProCH** [comment]
RTX 3080Ti   
Driver version: 31.0.15.5123  
![image](https://github.com/GTNewHorizons/Angelica/assets/29896317/c625d96c-bba5-4873-abc6-aa062a13ad69)

**Comment by mitchej123** [comment]
This should be fixed with alpha27
